frontend/pages/DashboardPage/cards/ActivityFeed/ActivityFeed.tsx

```diff
@@ -35,6 +36,7 @@ const ActivityFeed = ({
   const [showShowQueryModal, setShowShowQueryModal] = useState(false);
   const [showScriptDetailsModal, setShowScriptDetailsModal] = useState(false);
   const queryShown = useRef("");
+  const queryImpact = useRef<string | undefined>(undefined);
```

Comment:
You might consider putting both of these in the same ref 
```ts
  type queryDetails = {
    querySql: string;
    impact: string | undefined;
  };

const queryDetailsRef = useRef({ querySql: "", impact: undefined })<queryDetails>;
```

Although it sort of illustrates how the ref here is functioning as the props for the modal component, which makes me think there's probably a better pattern to achieve what we're trying to do here. Since we already have this pattern in place and it seems to be working ok, I don't think we need to change it up in this PR. Just raising it as food for thought.